_appmap/instrument.py

```diff
@@ -15,11 +15,12 @@
 @contextmanager
 def recording_disabled():
     tls = appmap_tls()
+    original_value = is_instrumentation_disabled()
```

Comment:
Seeing this led me to think a bit more about these changes.

In the work we've been going to generate large numbers of AppMaps, it's become clear that there's a significant performance penalty to running with the agent. We'll certainly need to take other steps to address this, but we it's something we should start thinking about.

Here, rather than calling `is_instrumentation_disabled()`, which will incur 4 or 5 extra function calls, let's just do
```suggestion
    original_value = tls.get("instrumentation_disabled")
```

Let's also remove `with recording_disabled():` from `describe_value`, and add it to `FuncReturnEvent.__init__`. It's already [here](https://github.com/getappmap/appmap-python/blob/60f3d145a5e8c65af202a8e03df509a3253f4947/_appmap/instrument.py#L85), so there's no need to call it again when creating `call` event.